spec/haproxy/templates/haproxy_config/frontend_https_spec.rb

```diff
@@ -178,68 +161,184 @@
     end
   end
 
-  context 'when ha_proxy.forwarded_client_cert is always_forward_only (the default)' do
-    it 'deletes the X-Forwarded-Client-Cert header by default' do
-      expect(frontend_https).to include('http-request del-header X-Forwarded-Client-Cert')
-    end
-  end
+  describe 'ha_proxy.forwarded_client_cert' do
+    context 'when ha_proxy.forwarded_client_cert is always_forward_only' do
+      let(:properties) do
+        default_properties.merge({ 'forwarded_client_cert' => 'always_forward_only' })
+      end
 
-  context 'when ha_proxy.forwarded_client_cert is forward_only' do
-    let(:properties) do
-      default_properties.merge({ 'forwarded_client_cert' => 'forward_only' })
-    end
+      it 'does not delete mTLS headers' do
+        expect(frontend_https).not_to include('http-request del-header X-Forwarded-Client-Cert')
+        expect(frontend_https).not_to include('http-request del-header X-SSL-Client')
```

Comment:
could these also be cut short using
```
expect(frontend_https).not_to include(/http-request del-header X-SSL-Client/)
```
?

Reply:
yep!